### PR TITLE
Run `cargo spellcheck` over every internal crate

### LINKED ISSUE #1672: Use `cargo-spellcheck` to automatically find typos in API documentation
**Issue summary**

The [cargo-spellcheck](https://crates.io/crates/cargo-spellcheck) command should be run in CI to catch typos in our API documentation. Enabling this check will require a lot of changes. This issue will track progress towards that goal.

To install `cargo-spellcheck`, run:
```bash
$ cargo install --locked cargo-spellcheck
```

General advice for updating the API documentation:
- Avoid mentioning data structures such as `Vec` or `HashMap` in the documentation unless it's absolutely required. Instead talk about vectors or mappings.
- When making references to data structures, mark them as code. Like this: `code`. Not this: code.
- As a last resort, add new words to the `.config/forest.dic` dictionary.
- To run spellcheck in a single crate or folder, enter the folder and run: `cargo spellcheck --cfg ../.config/spellcheck.toml`

**Other information and links**

Typos fixed as of July 27th: 13.

Examples:
- `unsigend` → `unsigned`
- `timetamp` → `timestamp`
- `Enviromental` → `Environmental`
- `Coeffiencients` → `Coefficients`
- `doesnt` → `doesn't`
- `orginating` → `originating`
- `paramaterized` → `parameterized`

### PR DESCRIPTION
**Pull request by lemmih**
**Summary of changes**
Changes introduced in this pull request:
- Fix a few remaining documentation issues.
- Remove duplicate and unused words from the custom dictionary.
- Run `cargo spellcheck` for the entire workspace.

**Reference issue to close (if applicable)**

Closes #1672

**Other information and links**

### CODE REVIEW

File: ipld/legacy_amt/src/amt.rs [hunk 1]
```diff
@@ -68,7 +68,7 @@ where
         self.root.bit_width
     }
 
-    /// Constructs an AMT with a blockstore and a Cid of the root of the AMT
+    /// Constructs an AMT with a block store and a Cid of the root of the AMT
```
Comment (elmattic) [inline ipld/legacy_amt/src/amt.rs:71]: Use `CID` here?

File: ipld/legacy_amt/src/amt.rs [hunk 2]
```diff
@@ -93,7 +93,7 @@ where
         self.root.count
     }
 
-    /// Generates an AMT with block store and array of cbor marshallable objects and returns Cid
+    /// Generates an AMT with block store and array of objects and returns Cid
```
Comment (elmattic) [inline ipld/legacy_amt/src/amt.rs:96]: Use `CID` here?

**Review by elmattic — DISMISSED** [review]
lgtm

File: .config/forest.dic [hunk 3]
```diff
@@ -1,52 +1,47 @@
...
+cron
+crypto
 Filecoin
+HAMT
+IPFS
 IPLD
 JSON
 JWT
+Kademlia
+LibP2P
```
Comment (LesnyRumcajs) [inline .config/forest.dic:24]: let's follow the official libp2p docs convention https://docs.libp2p.io/
```suggestion
libp2p
```

File: .config/forest.dic [hunk 4]
```diff
@@ -1,52 +1,47 @@
...
-MDNS
-KV
-Kademlia
-LibP2P
+synchronizer
+tipset/SM
+tipsetkey/S
 TOML
-bitfield
+Trie
```
Comment (LesnyRumcajs) [inline .config/forest.dic:43]:
```suggestion
trie
```

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**